compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -979,6 +979,18 @@ impl<'a> Parser<'a> {
                 // we are using noexpect here because we don't expect a `.` directly after a `return`
                 // which could be suggested otherwise
                 self.eat_noexpect(&token::Dot)
+            } else if self.token.kind == TokenKind::RArrow
+                && self.prev_token.span.overlaps_or_adjacent(self.token.span)
```

Comment:
In any case, could you remove this? It doesn't matter that the output for `tests/ui/parser/issues/issue-118530-ice.rs` changes. It's a regression test for an unrelated and fixed attribute recovery ICE. It's auto generated by a fuzzer. Just update the error annotations and `--bless` stderr, then we should be good to go.

Reply:
I was worried about modifying the output for this test, actually :sweat_smile: 

This should be fixed. Thanks!